src/Core/Services/BuildRequestStateMiddleware.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.DataApiBuilder.Core.Authorization;
+using HotChocolate.Execution;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using RequestDelegate = HotChocolate.Execution.RequestDelegate;
+
+/// <summary>
+/// This request middleware will build up our request state and will be invoke once per request.
+/// </summary>
+public sealed class BuildRequestStateMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public BuildRequestStateMiddleware(RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    public async ValueTask InvokeAsync(IRequestContext context)
+    {
+        if (context.ContextData.TryGetValue(nameof(HttpContext), out object? value) &&
+            value is HttpContext httpContext)
```

Comment:
no client role header key/value pair is added to the hot chocolate request context. Since the HC middleware pipeline is invoked after DAB's middleware, http context would be present with client role header. 

If a situation arises where no http context exists, then any attempt to retrieve client role header from HC request context will result in no value -> failed AuthZ.